### ISSUE TITLE
allow reloading banner following changes on WordPress (e.g. menu update)

### ISSUE THREAD

**Issue by denislepage** [MotusDev/Motus-TO-DO#70]
On the admin tools page, https://motus.org/data/admin, I wanted to see a link that allows refreshing the banner taken from the WordPress section of the site and kept in the application context, without having to do a server restart.

Please review the following steps, already completed (one remaining on TODO list) and ensure that my understanding of the structure is appropriate. The following changes have been posted to the sandbox branch:

1) I modified org.bsc.motus.web.SetupFilter.init() by moving the code loading the banner to a new static method of Attributes (loadWPHeaders), 

2) The static headers object (containing multiple languages) is no longer stored in SetupFilter, but is now in Attributes.

3) I added a call to loadWPHeaders() from SetupFilter.init()

4) My understanding of Attributes is that every session carried its own copy of the header in the appropriate language, creating multiple copies of the banner (one for each session). This appeared unnecessary duplication (?). Now that the headers object is stored in Attributes (see above), the getHeaderHtml method is no longer session specific and returns the header in the appropriate language (based on the session language). The setHeaderHtml was also made redundant and has been removed.

5) I modified AdminTools to allow calling Attributes.loadWPHeaders with a /headers_reload call, which I also added to the list of links available.

TODO: add an error code confirming the reload at the top of the admin page when reloading page.

Currently testing changes in sandbox, but the banner is failing to load. Looking into it.

**Comment by denislepage** [comment]
Seems to be working now. I have reverted a couple changes above: the setHeaderHtml method is brought back to Attributes, and the call is added back to SetupFilter in order for it to be available through the session object. However, the getHeaderHtml header method continues to return the copy from the static object, which remains stored as part of Attributes.

I need to understand better how the filter has been setup, but it seems like it should be modified to draw from the static object (getHeaderHtml) and maybe not the session object.

The reload_headers call in AdminTools appears to be working also now, but will need an error (success) code added.

**Comment by Hussell** [comment]
Note that the session attribute technically doesn't have a copy of the appropriate header, it has a reference to the same string which is in the application attribute (Map<Language, String>). Java Strings are immutable, so there's normally only one copy of each String in memory.

Nevertheless, I've changed the session attribute (a String containing the language-specific header for the current session) to an application attribute (Map<Language, String>), and made all the necessary alterations to get it to work. The static Map has been replaced by the application attribute, and the places in the JSP pages which used to say ${headerHtml} now say ${headersHtml[lang]}. This will hopefully look clearer and more obviously correct to the next person to view the code, a few months from now.

I've also added success and failure messages to the AdminTools manual reload.
